### DISCUSSION TITLE
Bypassing refetch using action data

### DISCUSSION THREAD

**Discussion by jwnx** [remix-run/remix#3176] [Q&A]
Hello, everyone.

As the [Remix documentation for Data Writes](https://remix.run/docs/en/v1/guides/data-writes) points out, when submitting a Form, Remix first calls the `action()` and then reloads all the data on the page. There are cases in which a database table update returns the updated object as response. The `action()` function, therefore, can have access to the most updated data, making a new data fetch unnecessary.

I'd like to know if there is a recommended way to deal with situations like these. I understand that I can use the `useActionData` hook to access `action()`'s data, but I would still need to use `useLoaderData()` for the first fetch. I also understand that reload can be optimized via [unstable_shouldreload](https://remix.run/docs/en/v1/api/conventions#unstable_shouldreload), but I wonder if this is our only option.

Thank you very much for your time

**Comment by kiliman** [comment]
Data revalidation is one of Remix's best features. It ensures that your UI is always up-to-date after mutations. Remember, you may have other data in parent layouts that are affected by this mutation.

Instead of returning the updated data from your action, I would simply redirect to the new page and let Remix do its thing. This way it works well even for non-JS: POST-REDIRECT-GET.

I generally only return data from action for validation errors, since I want the user to stay on the current route and the errors can be displayed in context. But I redirect on success.

Anyway, `unstable_shouldReload` is the escape hatch you can use if you really need to.
